source/includes/_rendering.md

```diff
@@ -178,4 +177,94 @@ function myRenderOverlay() {
\ No newline at end of file
+Then, for each part we render, we get half the width of the window, and then subtract half the width of our rendered object. For a string, this is done with `(width / 2) - (Renderer.getStringWidth(textToRender) / 2)`. For a fixed width object, you can replace `Renderer.getStringWidth(textToRender)` with the width of the object. Notice how we use the `Text` object and instantiate it outside of the render function. The `Text` object allows you to set additional properties, such as shadow and color.
+
+# Tessellator
+
+The Tessellator object is meant for rendering in 3 dimensions. You have to call this object under the `renderWorld` trigger.
+
+## Drawing Text
+
+> You can set floating text at specific world coordinates using `Tessellator.drawString(text, x, y, z)`
+
+```javascript
+register("renderWorld", myWorldRender);
+
+function myWorldRender() {
+  Tessellator.drawString("Hello there!", 100, 100, 100);
+}
+```
+
+This will draw a the words "Hello There" at the world location 100, 100, 100.
+You can also change the text color, whether or not to render the shadow, change the scale, and whether or not
+the size increases from distance. All of these examples below would work also.
+
+```javascript
+Tessellator.drawString("Blue text!", 100, 100, 100, Renderer.BLUE);
+Tessellator.drawString("Green text with shadow!", 100, 100, 100, Renderer.GREEN, true);
+Tessellator.drawString("Bigger red text!", 100, 100, 100, Renderer.RED, false, 2);
+Tessellator.drawString("Hello there!", 100, 100, 100, Renderer.DARK_PURPLE, true, 2, false);
+```
+
+## Drawing Textures
+
+> You can also draw specific textures like so
+
+```javascript
+register("renderWorld", myWorldRender);
+
+var img = new Image("kirbybit.png", "https://www.chattriggers.com/assets/images/kerbybit.png");
+
+function myWorldRender() {
+  Tessellator.bindTexture(img);
+  Tessellator.begin()
+    .translate(Player.getX(), Player.getY(), Player.getZ())
+    .pos(-0.5, 0.5, -0.5).tex(0, 0)
+    .pos(-0.5, 0.5, 0.5).tex(0, 1)
+    .pos(0.5, 0.5, 0.5).tex(1, 1)
+    .pos(0.5, 0.5, -0.5).tex(1, 0)
+    .draw();
+}
+```
+
+This will draw Kirbybit's skin texture on the player's feet position. This works by first binding the texture
+of the image to the `Tessellator` object. Then, it begins the instructions on what to do,
+by moving the center of the image to the player's position. Then, it has the 4 positions
+for each vertex, with the texture starting in the top left (0, 0) and working its way counter-clockwise.
+Then it's ended with `draw()` to finalize the texture.
```

Comment:
Might as well add a bit more detail here.

```suggestion
This will draw kerbybit's skin texture at the player's feet position. This works by first binding the texture
of the image to the `Tessellator` object. It then associates positions in the 3D world (relative to the
`Tesselator`'s current position) to points on the texture. The `tex` function takes an `x` and
`y` argument, both ranging from `0.0` to `1.0`, which identifies a point on the texture (i.e. 
`.tex(0.3, 0.7)` refers to the point 30% across from the left and 70% down from the top).
Finally, a call to `.draw()` finished the operation.
```